### ISSUE TITLE
Ending a drag of any kind causes a click event on whatever morph the cursor is over when released

### ISSUE THREAD

**Issue by liam4** [jmoenig/Snap--Build-Your-Own-Blocks#1586]
Hopefully these two gifs well demonstrate the issue:

![oops](https://cloud.githubusercontent.com/assets/9948030/21512538/fba41430-cc84-11e6-9530-a63ba6ba5649.gif)

![oops2](https://cloud.githubusercontent.com/assets/9948030/21512540/0394c68a-cc85-11e6-9153-3bbff3d64a0b.gif)

When I drag my mouse just about anywhere in the Snap<i>!</i> UI I've tried, releasing the mouse fires some kind of click event on the morph that's under it. The gifs demonstrate why this is a problem - when I'm scrolling by dragging, letting go triggers whatever block is below it; and resizing the category pane will focus on whatever category my mouse is over when I let go (if any).

**Comment by jmoenig** [comment]
Yes, this has been mentioned quite a few times, and it's clearly a misfeature that's on my list to address when I get around to it.

**Comment by DarDoro** [comment]
HandMorph.prototype.processMouseDown @ morphic.js unconditionally fires "mouseClick" event for any morph at cursor. 
"mouseClick" should be called ONLY after consecutive "mouseDown", "mouseUp" for the same morph. For example - morphAtCursor can be stored at "mouseDown" and tested in "mouseUp".

```
HandMorph.prototype.processMouseDown = function (event) {
...
        this.clickCandiateMorph = morph;


HandMorph.prototype.processMouseUp = function () {
...
        if(this.clickCandiateMorph){
            if(this.clickCandiateMorph == morph){
                morph[expectedClick](this.bounds.origin);`

```
